common/main.c (main_loop): new flag CONFIG_DONT_REPEAT to suppress automatic
  repetition of previous command
include/configs/neo1973.h: added CONFIG_DONT_REPEAT

Index: u-boot/common/main.c
===================================================================
--- u-boot.orig/common/main.c
+++ u-boot/common/main.c
@@ -481,7 +481,12 @@ void main_loop (void)
 			reset_cmd_timeout();
 		}
 #endif
+#ifdef CONFIG_DONT_REPEAT
+		do len = readline (CFG_PROMPT);
+		while (!len);
+#else
 		len = readline (CFG_PROMPT);
+#endif /* !CONFIG_DONT_REPEAT */
 
 		flag = 0;	/* assume no special flags for now */
 		if (len > 0)
Index: u-boot/include/configs/neo1973.h
===================================================================
--- u-boot.orig/include/configs/neo1973.h
+++ u-boot/include/configs/neo1973.h
@@ -215,6 +215,7 @@
 #define CFG_ENV_SIZE		0x4000		/* 16k Total Size of Environment Sector */
 #define CFG_ENV_OFFSET_OOB    1               /* Location of ENV stored in block 0 OOB */
 #define	CFG_PREBOOT_OVERRIDE	1	/* allow preboot from memory */
+#define CONFIG_DONT_REPEAT		/* don't repeat commands */
 
 #define NAND_MAX_CHIPS		1
 #define CFG_NAND_BASE		0x4e000000
